### PR TITLE
Bugfixes to CoreBase

### PR DESCRIPTION
**Pull request by theiostream**

### CODE REVIEW

**Review by stefanbidi — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+obj
+*.log
+*.sum
+GNUmakefile
```
Comment (fredkiefer) [inline .gitignore:4]: Why did you add GNUmakefile to .gitignore. Wont that result in all changes to the make file being lost?
